ccaas_builder/cmd/detect/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var logger = log.New(os.Stderr, "", 0)
+
+func main() {
+	logger.Println("::Detect")
+
+	if err := run(); err != nil {
+		logger.Printf("::Error: %v\n", err)
+		os.Exit(1)
+	} else {
+		logger.Printf("::Type detected as external")
+	}
+}
+
+type ChaincodeMetadata struct {
+	Type string `json:"type"`
+}
+
+func run() error {
+	if len(os.Args) > 2 {
+		chaincodeMetaData := os.Args[2]
+		mdbytes, err := ioutil.ReadFile(filepath.Join(chaincodeMetaData, "metadata.json"))
+		if err != nil {
+			return err
+		}
+
+		var metadata ChaincodeMetadata
+		err = json.Unmarshal(mdbytes, &metadata)
+		if err != nil {
+			return err
+		}
+
+		switch strings.ToLower(metadata.Type) {
+		case "external":
```

Comment:
if we only have one condition here... why not if else?

Reply:
a good question... we also need a better name not external

Reply:
- `CCAAS` ? 
- `CCaaS` ? 
- `ccaas` ? 
- `external-service` ? 
- `external-ccaas` ? 
- `external-chaincode` ?
- `service` ? 


(I like `ccaas`, despite it being _impossible_ to type into a form with an autocorrect spellchecker.)

Reply:
I'll update to use **ccaas** :-)